net/netns/netns_darwin.go

```diff
@@ -34,15 +42,118 @@ func controlLogf(logf logger.Logf, network, address string, c syscall.RawConn) e
 		// Don't bind to an interface for localhost connections.
 		return nil
 	}
-	idx, err := interfaces.DefaultRouteInterfaceIndex()
+
+	host, _, err := net.SplitHostPort(address)
```

Comment:
The splitting/parsing/validation wasn't being done before. For both performance and correctness/paranoia reasons you may want to keep this only inside the branch for the new code path.